### ISSUE TITLE
Add filter for assignee

### ISSUE THREAD

**Issue by NereusWB922** [CATcher-org/WATcher#342]
**Is your feature request related to a problem? Please describe.**
Currently, we can group issues and PRs by different criteria, not only by assignee. We should provide a filter for the assignee so users can filter the issue based on the assignee in different groupings.

**Comment by kokerinks** [comment]
hi can i work on this issue? will see if there's an intuitive way to add the filter for assignees as stated in the issue

**Comment by NereusWB922** [comment]
Sure, please go ahead and lmk if u have any question.

**Comment by kokerinks** [comment]
@NereusWB922 I have created the PR above as a prepatory PR to the enhancement.

If this is PR okay I intend to use `this.assignees` in the new `AssigneeService` to create the filter options, and will attempt to follow the format which the milestone filter was implemented.

**Comment by NereusWB922** [comment]
@kokerinks, Thanks for the pr. I will only be available to review prs during the weekend. If you are confident enough, you can proceed to implement the remaining logic. Just remember to checkout a new branch from ur  feature branch in case any changes are required for #379.

**Comment by kokerinks** [comment]
@NereusWB922 I have managed to get a preliminary assignee filter working, but I'd like to clarify on the expected behaviour here

Would you like the filter to filter based on the list of assignees in each issue/PR, or is it instead based on creator of each issue/PR?

My current implementation filters based on the former, and the behaviour doesnt seem to be what is expected:
<img width="1710" alt="image" src="https://github.com/CATcher-org/WATcher/assets/53986507/a77f2b34-3158-4510-a11e-126f79997066">
in this case, the list does show only unassigned issues/PRs, but there are still entries on some team members as they are unassigned issues who still have a creator.

**Comment by NereusWB922** [comment]
Hi @kokerinks,

Currently, in WATcher, we mainly focus on visualizing the feature/enhancement contributions of students to the project. This filter is primarily for tutors or other team members to easily identify the issues fixed by the students and the PRs created by the students.

I agree that the term "assignee" is a bit confusing as it refers to different things for GitHub Issues and PRs. Another alternative I considered when creating this issue was "Contributor," but that can also be misleading. If you have any recommendations, please suggest them.

To summarize again, the filtering logic should be:

- Filter by assignee for Issues
- Filter by author for PRs

PS (not related to this issue):
I personally think it would also be beneficial if we could see the PRs reviewed by a certain person or the issues created by a certain person. However, we would need a way to differentiate these contributions in the same view. You can create an issue regarding this if you have any ideas or to brainstorm with other developers.

**Comment by damithc** [comment]
> I agree that the term "assignee" is a bit confusing as it refers to different things for GitHub Issues and PRs. Another alternative I considered when creating this issue was "Contributor," but that can also be misleading. If you have any recommendations, please suggest them.

owner?

> I personally think it would also be beneficial if we could see the PRs reviewed by a certain person or the issues created by a certain person. However, we would need a way to differentiate these contributions in the same view. You can create an issue regarding this if you have any ideas or to brainstorm with other developers.

Yes, this can be useful.

**Comment by kokerinks** [comment]
> I personally think it would also be beneficial if we could see the PRs reviewed by a certain person or the issues created by a certain person. However, we would need a way to differentiate these contributions in the same view. You can create an issue regarding this if you have any ideas or to brainstorm with other developers.

Okay, will keep in mind for further directions after this PR.

> I agree that the term "assignee" is a bit confusing as it refers to different things for GitHub Issues and PRs. Another alternative I considered when creating this issue was "Contributor," but that can also be misleading. If you have any recommendations, please suggest them.

I ran this through GPT for suggestions, another possible one could be `IssueHandler`, or `TaskOwner` (similar to what @damithc suggested). Here's the chat for other suggestions provided by GPT: https://chatgpt.com/share/210ed050-fd19-4718-bd22-e0757a35d4dc

In the meanwhile, I will work off this and attempt to make the working filter first 
>To summarize again, the filtering logic should be:
> 
> Filter by assignee for Issues
> Filter by author for PRs

**Comment by NereusWB922** [comment]
Then, lets go with "TaskOwner" for now.

**Comment by kokerinks** [comment]
> Then, lets go with "TaskOwner" for now.

Okay, will take note.

In the meanwhile, I've managed to get the filter working as intended, butwith a minor issue: it does not correctly filter issues with multiple taskowners. 

For example:
<img width="1030" alt="image" src="https://github.com/user-attachments/assets/85b9b7c6-a0ce-449f-9b3e-ae0051cb1e2d">

In this case, the `kokerinks` column should not be showing as it is not selected in the filter. However, one issue (number 180) is still displayed under the `kokerinks` column as the issue is tagged to both `billyhoce` and `kokerinks`:
![image](https://github.com/user-attachments/assets/8e9380cb-79a7-4cc4-be91-7b5e7b3ab8bc)

This issue involves changing how group-handling is currently implemented. Hence, is it possible to review and merge in this PR first, then create a new issue for the problem stated above? I intend to refactor the groups functionality (and possibly the filters as well) before attempting to fix this issue

Of course, if you have any other suggestions on how I should approach this, feel free to let me know too! am open to suggestions

I have pushed the implementation to PR #381, can take a look and test around
